tasks.py

```diff
@@ -26,3 +28,42 @@ def _migrate_file(url: str, filename: str):
 
     _migrate_file(sources_url, "db_sources.csv")
     _migrate_file(signal_url, "db_signals.csv")
+
+
+@task
+def lint(
+    c, incremental=True, format=True, revision="origin/dev...", diff=False, check=False
+):  # pylint: disable=redefined-builtin
+    """Lint and format.
+
+    Additional linter settings can be found in `pyproject.toml` (this invocation
+    will use those settings as well).
+
+    Parameters
+    ----------
+    incremental : bool
+        Only lint changed files.
+    format : bool
+        Apply formatting changes.
+    revision : str
+        Revision to compare against.
+    diff : bool
+        Only show formatting changes, do not apply.
+    """
+    diff = "--diff" if diff else ""
+    check = "--check" if check else ""
+    if incremental:
+        if format:
+            c.run(f"darker --revision {revision} {diff} {check} .")
+        out = c.run(f"git diff -U0 {revision} | lint-diffs")
+        if out.stdout.strip() != "=== pylint: mine=0, always=0":
+            print(out.stdout)
+            sys.exit(1)
+    else:
+        if format:
+            reponse = input("This will format all files in this repo, continue? [y/N]")
```

Comment:
scary!  nice failsafe :)

Reply:
Thanks! I must confess that this in here because I already shot myself with this footgun xD